jupyter/nmm-minimax.ipynb

```diff
@@ -40,6 +41,34 @@
     "Logischerweise ist *Minimax* somit nicht für die komplette Berechnung des Spiels geeignet, weil dies die üblicherweise zur Verfügung stehenden Ressourcen überschreitet. Aus diesem Grund wird die Tiefensuche auf eine maximale Tiefe `limit` beschränkt. Dies hat jedoch zur Folge, dass bei der maximalen Tiefe häufig noch kein eindeutiges Ergebnis *Mini* dem Wert `-1` (sichere Niederlage für `player`) oder *Max* mit dem Wert `1` (sicherer Sieg für `player`) erkannt werden konnte. Somit ist die Funktion `heuristic(state, player)` notwendig die in einem solchen Fall eine einfache heuristische Bewertung des `state` für den `player` durchführt und einen Wert $ -1 < value < 1 $ berechnet."
    ]
   },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Für die Implementierung des *Minimax* Algorithmuses wird nun eine Klasse `Minimax` implementiert, die von der zuvor definierten Klasse `ArtificialIntelligence` erbt. Hierzu wird die Funktion `bestMoves` überschrieben, die später genauer definiert wird, und der Konstruktor `__init__` implementiert. Der Konstrukter besitzt zwei optionale Parameter, die den Algorithmus konfigurieren können:\n",
+    "\n",
+    "* Die `limit` Einstellung setzt die maximale Rekursionstiefe;\n",
+    "* Durch den `weights` Parameter können die Gewichtungen der zuvor definierten Heuristik kontrolliert werden.\n",
+    "\n",
+    "Zusätzlich initialisiert der Konstruktor das Attribut `cache` mit einem leeren `dict`. Dieses wird später als Transpositionstabelle verwendet."
```

Comment:
Wurde Transpositionstabelle vorher schonmal genauer erklärt? Weil Google sagt [komische Sachen](https://www.delamar.de/songwriting/kostenlose-transpositionstabelle-zum-download-mit-anleitung-4751/) dazu xD.

Reply:
"Die Hash-Tabelle von bereits zuvor gesehenen Positio-nen  wird  traditionell  als  Transpositionstabelle  bezeichnet;  sie  ist  im  Wesentlichen  identisch  mit  der  untersuchten  Liste  in  GRAPH-SEARCH (Abschnitt  3.3).  Die  Verwen-dung einer Transpositionstabelle kann drastische Auswirkungen haben – beim Schach manchmal bis zu einer Verdopplung der erreichbaren Suchtiefe."
https://vowi.fsinf.at/images/b/bc/TU_Wien-Einf%C3%BChrung_in_die_K%C3%BCnstliche_Intelligenz_VU_%28Eiter%2C_Tompits%29_-_K%C3%BCnstliche_Intelligenz-_Ein_moderner_Ansatz_%283.%2C_aktualisierte_Auflage%29.pdf

Reply:
Müsste korrekt sein :D